Products/CMFPlone/browser/login/password_reset.py

```diff
@@ -90,14 +90,15 @@ class PasswordResetView(BrowserView):
 
     def _auto_login(self, userid, password):
         aclu = getToolByName(self.context, 'acl_users')
+        membership_tool = getToolByName(self.context, 'portal_membership')
         for name, plugin in aclu.plugins.listPlugins(ICredentialsUpdatePlugin):
             plugin.updateCredentials(
                 self.request,
                 self.request.response,
                 userid,
                 password
             )
-        user = getSecurityManager().getUser()
+        user = membership_tool.getMemberById(userid).getUser()
         login_time = user.getProperty('login_time', None)
         if login_time is None:
             notify(UserInitialLoginInEvent(user))
```

Comment:
If the notify still needed or does `getMemberById` takes care of tit as well?

Reply:
@jensens I think **notify** is still needed here because **getMemberById** -- or the **updateCredentials** above -- is not authenticating the user.

The original code would always, in a default Plone installation, return an Anonymous User,  and that is why we use the membership tool to get a user object for that userid.

Reply:
Not related to this PR, but it feels strange to me that a call to `getSecurityManager().getUser()` in a method called `_auto_login` before firing the `UserLoggedInEvent` does not return the proper user :thinking:

Reply:
@ale-rt I agree. My understanding of the whole situation is the following:

- Request to the password reset view is done without a valid auth cookie
- Zope/PAS authentication considers this user as anonymous
- After the password is reset we come to this method, still without authentication
- We call update credentials for all registered plugins (in the default Plone installation, this will send back a new cookie to the user)
- Current request still has no auth cookie (only the response will set it)
- getSecurityManager still believes this is an Anonymous user
- User receives a redirect and the new cookie
- Next request will see the user authenticated (as the new cookie is valid)

Reply:
Yes, that is really plausible. I am wondering if this method should actually login the user so that the security manager is aware of that.
But maybe this has some other implications.
For me the PR is fine as a workaround (then mergeable) but the real issue is that the user at this point is not logged in.
Any other opinion?